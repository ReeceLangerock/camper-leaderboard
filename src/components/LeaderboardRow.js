import React from "react";
import styled from "styled-components";

const Image = styled.img`
  max-height: 40px;
  width: 40px;
  margin-right: 5px;
  border-radius: 3px;
  border: 1px solid green;
  box-sizing: border-box;
`;

const Col = styled.td`
  padding: 5px;
  border-right: 1px solid;
`;
const Camper = Col.extend`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  text-align: left;

`;

const Row = styled.tr`
  &:nth-child(even) {
    background-color: #DEE7E7;
  }
`;

export class LeaderboardRow extends React.Component {
  render() {
    const camper = this.props.camper;
    return (
      <Row>
        <Col key ={'rank'} >{this.props.rank}</Col>
        <Camper key ={'camper'}>
          <Image src={camper.img} /> {camper.username}
        </Camper>
        <Col key ={'alltime'}>{camper.alltime}</Col>
        <Col key ={'recent'}>{camper.recent}</Col>
      </Row>
    );
  }
}
export default LeaderboardRow;
