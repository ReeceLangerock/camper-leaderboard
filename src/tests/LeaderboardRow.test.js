import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, find, render, simulate } from "enzyme";
import LeaderboardRow from "./../components/LeaderboardRow";
import enzymeSerializer from "enzyme-to-json/serializer";

expect.addSnapshotSerializer(enzymeSerializer);
describe("LeaderboardRow", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<LeaderboardRow />, div);
  });

  let component;
  beforeEach(() => {
    const props = {};
    component = mount(<LeaderboardRow {...props} />);
  });

  it("has the correct class", () => {
    expect(component.hasClass("LeaderboardRow")).toEqual(true);
  });
});

describe("Snapshot LeaderboardRow", () => {
  it("matches snapshot", () => {
    const comp = mount(<LeaderboardRow />);
    expect(comp).toMatchSnapshot();
  });
});
