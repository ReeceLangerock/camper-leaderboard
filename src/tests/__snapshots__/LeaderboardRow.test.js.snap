// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot LeaderboardRow matches snapshot 1`] = `
<LeaderboardRow>
  <div
    className="LeaderboardRow"
  />
</LeaderboardRow>
`;
